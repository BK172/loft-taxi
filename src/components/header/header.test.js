import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Header from './header';

describe('Header component test', () => {
  it('Should render Header correctly', () => {
    const tree = renderer.create(
      <Header
        onPageChange={jest.fn()}
        logIn={jest.fn()}
        logOut={jest.fn()}
        isLoggedIn={true}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Click on navigation button', () => {
    const onPageChange = jest.fn();
    const wrapper = shallow(
        <Header
          onPageChange={jest.fn()}
          logIn={jest.fn()}
          logOut={jest.fn()}
          isLoggedIn={true}
        />
    );

    const buttons = wrapper.find('button');

    buttons.forEach((button) => button.simulate('click', {
      preventDefault: jest.fn(),
    }));

    expect(onPageChange).toHaveBeenCalledTimes(buttons.length);
  });
});