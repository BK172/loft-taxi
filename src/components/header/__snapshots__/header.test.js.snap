// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test Should render Header correctly 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root header MuiToolbar-regular MuiToolbar-gutters"
  >
    <div
      className="header__logo"
    >
      <img
        alt="LoftTaxi logo"
        src="logo-header.svg"
      />
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label header__button header__button_active"
      >
        Карта
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label header__button"
      >
        Профиль
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label header__button"
      >
        Войти
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label header__button"
      >
        Выйти
      </span>
    </button>
  </div>
</header>
`;
